cirq_superstaq/ops/qubit_gates.py

```diff
@@ -753,6 +753,75 @@ def _from_json_dict_(cls, **kwargs: Any) -> "IXGate":
 AQTICCX = AQTITOFFOLI = IX.controlled(2, [0, 0])
 
 
+@cirq.value_equality(approximate=True)
+class StrippedCZGate(cirq.Gate):
+    """The Stripped CZ Gate is the gate that is actually being performed by Hilbert,
+    which is just a regular CZ gate when phi = 0, and is corrected into a CZ
+    gate by RZ gates afterwards if phi is nonzero.
+    """
+
+    def __init__(self, phi: cirq.TParamVal = 0) -> None:
+        """Args: phi: CZ-rotation angle in radians"""
+        self.phi = phi
+
+    def _num_qubits_(self) -> int:
+        return 2
+
+    def _unitary_(self) -> Optional[npt.NDArray[np.complex_]]:
+        if self._is_parameterized_():
+            return None
+        return np.diag(
+            [1.0, np.exp(1j * self.phi), np.exp(1j * self.phi), np.exp(1j * (2 * self.phi - np.pi))]
+        )
+
+    def _value_equality_values_(self) -> cirq.TParamVal:
+        if cirq.is_parameterized(self.phi):
+            return self.phi
+        return self.phi % (2 * np.pi)
+
+    def _value_equality_approximate_values_(self) -> cirq.PeriodicValue:
+        return cirq.PeriodicValue(self.phi, 2 * np.pi)
+
+    def __pow__(self, exponent: cirq.TParamVal) -> "StrippedCZGate":
+        if exponent == 1:
+            return StrippedCZGate(exponent * self.phi)
+        return NotImplemented
+
+    def __str__(self) -> str:
+        return f"StrippedCZGate({self.phi})"
+
+    def __repr__(self) -> str:
+        return f"css.StrippedCZGate({self.phi})"
+
+    def _decompose_(self, qubits: Tuple[cirq.Qid, cirq.Qid]) -> Iterator[cirq.Operation]:
+        yield cirq.rz(self.phi).on(qubits[0])
+        yield cirq.rz(self.phi).on(qubits[1])
+        yield cirq.CZ(*qubits)
+
+    def _resolve_parameters_(
+        self, resolver: cirq.ParamResolver, recursive: bool
+    ) -> "StrippedCZGate":
+        return StrippedCZGate(
+            cirq.resolve_parameters(self.phi, resolver, recursive),
+        )
+
+    def _is_parameterized_(self) -> bool:
+        return cirq.is_parameterized(self.phi)
+
+    def _parameter_names_(self) -> AbstractSet[str]:
+        return cirq.parameter_names(self.phi)
+
+    def _has_unitary_(self) -> bool:
+        return not self._is_parameterized_()
+
+    def _circuit_diagram_info_(self, args: cirq.CircuitDiagramInfoArgs) -> cirq.CircuitDiagramInfo:
+        p = args.format_radians(self.phi)
+        return cirq.CircuitDiagramInfo(wire_symbols=("@", f"@({p})"))
```

Comment:
nit (previously discussed offline, but adding here so we don't forget):
```suggestion
        return cirq.CircuitDiagramInfo(wire_symbols=("@({p})", f"@({p})"))
```
It's technically duplicate info, but it signals that both qubits are modified by an Rz gate.